### ISSUE TITLE
Add usage data from API response in StreamResponseFromChatbotAsync

### ISSUE THREAD

**Issue by johnkm516** [OkGoDoIt/OpenAI-API-dotnet#114]
Per the [OpenAI documentation](https://platform.openai.com/docs/guides/chat), Chats, Text completions and embeddings contain usage data in this format : 

```
{
 'id': 'chatcmpl-6p9XYPYSTTRi0xEviKjjilqrWU2Ve',
 'object': 'chat.completion',
 'created': 1677649420,
 'model': 'gpt-3.5-turbo',
 'usage': {'prompt_tokens': 56, 'completion_tokens': 31, 'total_tokens': 87},
 'choices': [
   {
    'message': {
      'role': 'assistant',
      'content': 'The 2020 World Series was played in Arlington, Texas at the Globe Life Field, which was the new home stadium for the Texas Rangers.'},
    'finish_reason': 'stop',
    'index': 0
   }
  ]
}
```

However this data is not captured in the library. Usage data is extremely important to keep track of how much tokens each user is using up and perhaps put a cap on the amount of tokens per user.  

EDIT : 

I should specify, I'm specifically using StreamResponseFromChatbotAsync function to stream the response result. However this function only gives me the string response string inside the stream, and the return value is void.

**Comment by johnkm516** [comment]
Never mind.

The README doesn't specifically show 
```Chat.StreamChatAsync```
which returns the full API response and I missed it.

**Comment by OkGoDoIt** [comment]
If you’re using the conversation object, there is a separate property for
the entirety of the most recent response which you can use to access the
usage statistics as well

**Comment by johnkm516** [comment]
> If you’re using the conversation object, there is a separate property for the entirety of the most recent response which you can use to access the usage statistics as well

Thanks for the response, but the problem is on OpenAI's end, not with your API. 

After actually implementing with StreamChatAsync I noticed the ChatUsage object is null. I used breakpoints to check every iteration of the stream and it's null always. The conversation.MostResentApiResult.Usage is also null. 

So I looked around and found this [thread](https://community.openai.com/t/openai-api-get-usage-tokens-in-response-when-set-stream-true/141866). HTTP Stream requests via the OpenAI API returns null for usage. But it's easy to calculate, as the person in the thread says, every async call of the result handler in the stream is a token. So all I did was create a variable called  responseTokenCount and incremented it. The user token count is a different issue though, I will probably have to find a library for that.
